core/server/middleware/api/cors.js

```diff
@@ -33,9 +32,15 @@ function getIPs() {
 }
 
 function getUrls() {
-    var urls = [url.parse(utils.url.urlFor('home', true)).hostname];
+    var blogHost = url.parse(utils.url.urlFor('home', true)).hostname,
+        adminHost = url.parse(utils.url.urlFor('admin', true)).hostname,
+        urls = [];
 
-    // @TODO: add admin.url if available
+    urls.push(blogHost);
+
+    if (adminHost !== blogHost) {
+        urls.push(adminHost);
+    }
```

Comment:
If admin hostname is available and it's not the blog hostname, we push the hostname for whitelisted urls.